### ISSUE TITLE
App Crash on Updating to Latest Main (fd51f26ff6a4aa1e93bb865cff5b001b25565818)

### ISSUE THREAD

**Issue by qiarie** [opensrp/fhircore#2763]
**Describe the bug**

The application crashes on opening a questionnaire that has an expression with the language set as `application/x-fhir-query` after merging latest main - reverting to commit `595f422af2723b1993929965ad876172850d8f0c` resolves the issue.

Stack trace of the crash
````
FATAL EXCEPTION: main
Process: org.smartregister.opensrp.zeir, PID: 23231
java.lang.IllegalArgumentException: Unsupported expression language, language should be text/fhirpath
	at com.google.android.fhir.datacapture.fhirpath.ExpressionEvaluator.evaluateVariable(ExpressionEvaluator.kt:475)
	at com.google.android.fhir.datacapture.fhirpath.ExpressionEvaluator.evaluateQuestionnaireVariableExpression$datacapture_release(ExpressionEvaluator.kt:278)
	at com.google.android.fhir.datacapture.fhirpath.ExpressionEvaluator.evaluateQuestionnaireVariableExpression$datacapture_release(ExpressionEvaluator.kt:268)
	at com.google.android.fhir.datacapture.fhirpath.ExpressionEvaluator.findAndEvaluateVariable(ExpressionEvaluator.kt:418)
	at com.google.android.fhir.datacapture.fhirpath.ExpressionEvaluator.extractDependentVariables(ExpressionEvaluator.kt:230)
	at com.google.android.fhir.datacapture.fhirpath.ExpressionEvaluator.evaluateExpression(ExpressionEvaluator.kt:104)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel.resolveCqfExpression(QuestionnaireViewModel.kt:541)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireAdapterItems(QuestionnaireViewModel.kt:683)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireAdapterItems(QuestionnaireViewModel.kt:640)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel.getQuestionnaireState(QuestionnaireViewModel.kt:588)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel.access$getQuestionnaireState(QuestionnaireViewModel.kt:80)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel$questionnaireStateFlow$1.invokeSuspend(QuestionnaireViewModel.kt:478)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel$questionnaireStateFlow$1.invoke(Unknown Source:9)
	at com.google.android.fhir.datacapture.QuestionnaireViewModel$questionnaireStateFlow$1.invoke(Unknown Source:20)
	at kotlinx.coroutines.flow.FlowKt__ZipKt$combine$$inlined$combineUnsafe$FlowKt__ZipKt$1$2.invokeSuspend(Zip.kt:333)
	at kotlinx.coroutines.flow.FlowKt__ZipKt$combine$$inlined$combineUnsafe$FlowKt__ZipKt$1$2.invoke(Unknown Source:13)
	at kotlinx.coroutines.flow.FlowKt__ZipKt$combine$$inlined$combineUnsafe$FlowKt__ZipKt$1$2.invoke(Unknown Source:6)
	at kotlinx.coroutines.flow.internal.CombineKt$combineInternal$2.invokeSuspend(Combine.kt:76)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.EventLoop.processUnconfinedEvent(EventLoop.common.kt:68)
	at kotlinx.coroutines.DispatchedTaskKt.resumeUnconfined(DispatchedTask.kt:245)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:163)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:474)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:508)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:497)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:368)
	at kotlinx.coroutines.flow.SharedFlowImpl.tryEmit(SharedFlow.kt:399)
	at kotlinx.coroutines.flow.internal.SubscriptionCountStateFlow.increment(AbstractSharedFlow.kt:129)
	at kotlinx.coroutines.flow.internal.AbstractSharedFlow.allocateSlot(AbstractSharedFlow.kt:68)
	at kotlinx.coroutines.flow.StateFlowImpl.collect(StateFlow.kt:382)
	at kotlinx.coroutines.flow.ReadonlyStateFlow.collect(Unknown Source:2)
	at com.google.android.fhir.datacapture.QuestionnaireFragment$onViewCreated$4.invokeSuspend(QuestionnaireFragment.kt:143)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at androidx.lifecycle.DispatchQueue.drainQueue(DispatchQueue.kt:74)
	at androidx.lifecycle.DispatchQueue.resume(DispatchQueue.kt:53)
	at androidx.lifecycle.LifecycleController.observer$lambda$0(LifecycleController.kt:40)
	at androidx.lifecycle.LifecycleController.$r8$lambda$INWMOb0xmlhb_ycZo7uXu7mUdSw(Unknown Source:0)
	at androidx.lifecycle.LifecycleController$$ExternalSyntheticLambda0.onStateChanged(Unknown Source:4)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.kt:314)
	at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.kt:251)
	at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.kt:287)
	at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.kt:136)
	at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.kt:119)
	at androidx.fragment.app.FragmentViewLifecycleOwner.handleLifecycleEvent(FragmentViewLifecycleOwner.java:100)
	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:708)
	at androidx.fragment.app.Fragment.restoreViewState(Fragment.java:3176)
	at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:3161)
	at androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:619)
	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:275)
	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1943)
	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1845)
	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1782)
	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:565)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7872)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@96dda08, Dispatchers.Main.immediate]
````

**To Reproduce**
1. Run the ZEIR app
2. Go to Adults profile
3. Click on an adult without COVID vaccines administered
4. Go to the overflow menu and select Record Covid Vaccination
5. The app crashes

**Expected behavior**

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: [e.g. Samsung Galaxy A2 emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]
 - Version [e.g. 24]

**Additional context**

**Comment by FikriMilano** [comment]
I'm 99% sure it's because our SDC snapshot doesn't have this change.
https://github.com/google/android-fhir/pull/2076/files#diff-2c09e15c690a99edd91a802ba26f41b814d9b842081714451b1dc550eb25722eR427-R454

@qiarie can you merge that PR to our current SDC release branch? I've solved the merge conflicts. So you're good to go.
@ndegwamartin which branch should we merge that PR to?

**Comment by qiarie** [comment]
Resolved by issue [#2778](https://github.com/opensrp/fhircore/pull/2778)
